### DISCUSSION TITLE
Query regarding noise in HW aware training

### DISCUSSION THREAD

**Discussion by IannoIITR** [IBM/aihwkit#544] [Q&A]
As I see there are 2 types of noise in the rpu config definition the normal analog noises read, programming and drift. and the forward noises w_noise and out_noise. My query was reagarding how do these noises affect training? When I was running the example 5 the analog noises would get applied only during the inference stage and wouldn't affect calculated loss in any way, and even w_noise isn't affecting the accuracy significantly even when I increase its magnitude to large values. So I wanted to ask what noise plays a role in the training process for PCM tile? 
I was expecting something like backprop to be ideal but the forward pass during loss calculation would include some noise so that the gradient would be updated according to the noisy inference. Is this a wrong assumption?

**Comment by maljoras** [comment]
HI @IannoIITR ,
thanks for the question. Indeed, there is the possibility to inject noise for hardware-aware training. However, not all fields in the `InferenceRPUConfig` (see [here](https://github.com/IBM/aihwkit/blob/master/src/aihwkit/simulator/configs/configs.py#L146)) are applied during training, some are only used for simulating the inference on the hardware . In particular, the settings in `noise_model`, `drift_compensation` are *only* applied during inference to accurately model the deployment on the inference hardware. The other the fields,  ie. `modifier`, `clip` and `remap` set the weight noise injection and clipping and remapping strategy, respectively, during the hardware-aware training and are *only* relevant during training; these are probably the fields you are looking for. Finally, `forward`  specifies all short-term MAC nonidealities of the hardware. They are applied in *both* cases, hardware-aware training as well as during inference (same for `mapping` and `pre_post` as they define the DNN architecture and some chip assumptions). `forward` settings are another place where you can use noise injection (on inputs and outputs) during training, but be aware that the same noise is applied during inference as well (if you do not change it explicitly during inference to reflect your hardware noise properties).        

Please also take a look at the recent  [tutorial](https://arxiv.org/abs/2307.09357) that explains the hardware-aware training in more details.
